templates/category.tpl

```diff
@@ -8,7 +8,8 @@
 
 		<div class="clearfix">
 			<!-- IF privileges.topics:create -->
-			<button component="category/post" id="new_topic" class="btn btn-primary">[[category:new_topic_button]]</button>
+			<a href="{config.relative_path}/compose?cid={cid}" component="category/post" id="new_topic_no_script" class="btn btn-primary noscript">[[category:new_topic_button]]</a>
```

Comment:
I think these can be combined into a single element like so:
```
<a href="{config.relative_path}/compose?cid={cid}" component="category/post" id="new_topic" class="btn btn-primary">[[category:new_topic_button]]</a>
```

That should work as long as we correctly do the `event.preventDefault();`